chain/messagepool/messagepool.go

```diff
@@ -293,38 +303,41 @@ func (mp *MessagePool) Add(m *types.SignedMessage) error {
 func (mp *MessagePool) addTs(m *types.SignedMessage, curTs *types.TipSet) error {
 	// big messages are bad, anti DOS
 	if m.Size() > 32*1024 {
-		return xerrors.Errorf("mpool message too large (%dB): %w", m.Size(), ErrMessageTooBig)
+		return NewError(fmt.Sprintf("mpool message too large (%dB)", m.Size()), false)
 	}
 
 	if m.Message.To == address.Undef {
-		return ErrInvalidToAddr
+		return NewError("message had invalid to address", false)
 	}
 
 	if !m.Message.Value.LessThan(types.TotalFilecoinInt) {
-		return ErrMessageValueTooHigh
+		return NewError("cannot send more filecoin than will ever exist", false)
 	}
 
 	if err := sigs.Verify(&m.Signature, m.Message.From, m.Message.Cid().Bytes()); err != nil {
-		log.Warnf("mpooladd signature verification failed: %s", err)
-		return err
+		return NewError(fmt.Sprintf("mpooladd signature verification failed: %s", err), false)
 	}
 
 	snonce, err := mp.getStateNonce(m.Message.From, curTs)
 	if err != nil {
-		return xerrors.Errorf("failed to look up actor state nonce: %w", err)
+		return  NewError(fmt.Sprintf("failed to look up actor state nonce: %s", err), true)
 	}
 
 	if snonce > m.Message.Nonce {
-		return xerrors.Errorf("minimum expected nonce is %d: %w", snonce, ErrNonceTooLow)
+		return  NewError(fmt.Sprintf("minimum expected nonce is %d", snonce), false)
+	}
+
+	if snonce + bigNonceThreshold < m.Message.Nonce {
```

Comment:
This feels very arbitrary and will break the nonce-fix tool we use for fixing faucet from time to time.
IDK how I feel about it.

Reply:
Maybe the solution is to look at other messages in mpool to check if they fill the gap.

Reply:
I'm going to take it out for now. We'll probably have to keep stepping up our spam prevention as the "attacks" come in.